src/main.rs

```diff
@@ -1,45 +1,105 @@
-    let api = Api::new(token);
-
-    // Fetch new updates via long poll method
-    let mut stream = api.stream();
-    while let Some(update) = stream.next().await {
-        // If the received update contains a new message...
-        let update = update?;
-        println!("{:?}", update);
-        if let UpdateKind::Message(message) = update.kind {
-            if let MessageKind::Text { ref data, .. } = message.kind {
-                // Print received text message to stdout.
-                println!("<{}>: {}", &message.from.first_name, data);
-                if_chain! {
-	                if let Some(ref reply_to_message) = message.reply_to_message;
-	                if let MessageOrChannelPost::Message(mess) = &**reply_to_message;
-	                if mess.from.username.as_ref().unwrap_or(&"".to_owned()) == &bot_name;
-	                then {
-			                api.send(message.text_reply(vs[rand::thread_rng().gen_range(0..vs.len())]))
-			                .await?;
-	        	 	} else {
-		                if re.is_match(data) {
-			                api.send(message.text_reply(vs[rand::thread_rng().gen_range(0..vs.len())]))
-			                .await?;
-		                }
-	        	 	}
-	            }
-            }
+use clap::{App, SubCommand};
+use crate::toxic_bot::ToxicBot;
+use std::{fs, env};
+use std::fs::{File};
+use crate::constants::*;
+use std::error::Error;
+use std::io::{BufReader, BufRead};
+use whatlang::Lang;
+use crate::telegram_bot::TelegramBot;
+
+mod toxic_bot;
+mod cmd_bot;
+mod constants;
+mod telegram_bot;
+
+fn main() {
+    env_logger::init();
+
+    let matches = App::new("ToxicBot")
+        .about("A bot that will roast you")
+        .author("Bohdan Iakymets <cyberbond95@gmail.com>, Serhii Zakharov <serhii@zahar.pro>")
+        .subcommand(
+            SubCommand::with_name("cmd")
+                .about("just for testing, or if you're feeling \
+                        like you want to be roasted and don't have Telegram")
+        )
+        .subcommand(
+            SubCommand::with_name("telegram")
+                .about("to run telegram bot. \
+                        You'll also need to set TELEGRAM_BOT_TOKEN env variable")
+        )
+        .get_matches();
+
+    let mut bot = ToxicBot::new();
+
+    load_dir(&mut bot, DEFAULT_DATA_PATH).unwrap();
+
+    match matches.subcommand_name() {
+        Some("cmd") => cmd_bot::run(&mut bot),
+        Some("telegram") => {
+            let token = env::var("TELEGRAM_BOT_TOKEN")
+                .map_err(|_| "telegram bot token is not set")
+                .unwrap();
+            TelegramBot::new(&mut bot, token).unwrap().run().unwrap()
         }
+        _ => println!("{}", matches.usage()),
     }
+}
+
+fn load_dir(bot: &mut ToxicBot, path: &str) -> Result<(), Box<dyn Error>> {
```

Comment:
I think loading would be better move to separate module. Main will be more cleaner.

Reply:
moved them to the bot itself